Backport of the upstream fix that keeps redistributable components without
targets (cufile, nvidia_fs in CUDA 12.9) out of the toolkit's aggregate deps.
Drop once rules_cuda releases past 0.3.0.

--- a/cuda/private/template_helper.bzl
+++ b/cuda/private/template_helper.bzl
@@ -28,13 +28,16 @@
 def _expand_dctk_cuda(repository_ctx, components):
     tpl_label = Label("//cuda/private:templates/BUILD.dctk_cuda")
 
-    all_files_srcs_line = [comp + "_all_files" for comp in components.keys()]
+    # Filter out components with empty REGISTRY entries
+    valid_components = [comp for comp in components.keys() if len(REGISTRY[comp]) > 0]
+
+    all_files_srcs_line = [comp + "_all_files" for comp in valid_components]
     all_files_srcs_line = "srcs = " + repr(all_files_srcs_line)
 
-    license_srcs_line = [comp + "_license" for comp in components.keys()]
+    license_srcs_line = [comp + "_license" for comp in valid_components]
     license_srcs_line = "srcs = " + repr(license_srcs_line)
 
-    headers_deps_line = [comp + "_headers" for comp in components.keys()]
+    headers_deps_line = [comp + "_headers" for comp in valid_components]
     headers_deps_line = "deps = " + repr(headers_deps_line)
 
     substitutions = {
